core/src/main/java/com/linecorp/armeria/internal/metric/RequestMetricSupport.java

```diff
@@ -40,56 +40,64 @@
  */
 public final class RequestMetricSupport {
 
-    private static final AttributeKey<RequestMetrics> ATTR_REQUEST_METRICS =
-            AttributeKey.valueOf(RequestMetricSupport.class, "REQUEST_METRICS");
+    private static final AttributeKey<ActiveRequestMetrics> ATTR_ACTIVE_REQUEST_METRICS =
+            AttributeKey.valueOf(RequestMetricSupport.class, "ACTIVE_REQUEST_METRICS");
 
-    private static final RequestMetrics PLACEHOLDER = new RequestMetricsPlaceholder();
+    private static final ActiveRequestMetrics PLACEHOLDER = new ActiveRequestMetricsPlaceHolder();
 
     public static void setup(RequestContext ctx, MeterIdPrefixFunction meterIdPrefixFunction) {
-        if (ctx.hasAttr(ATTR_REQUEST_METRICS)) {
+        if (ctx.hasAttr(ATTR_ACTIVE_REQUEST_METRICS)) {
             return;
         }
 
         // Set the attribute to the placeholder so that calling this method again has no effect.
         // The attribute will be set to the real one later in onRequestStart()
-        ctx.attr(ATTR_REQUEST_METRICS).set(PLACEHOLDER);
+        ctx.attr(ATTR_ACTIVE_REQUEST_METRICS).set(PLACEHOLDER);
 
         ctx.log().addListener(log -> onRequestStart(log, meterIdPrefixFunction),
                               RequestLogAvailability.REQUEST_HEADERS,
                               RequestLogAvailability.REQUEST_CONTENT);
         ctx.log().addListener(RequestMetricSupport::onRequestEnd,
                               RequestLogAvailability.REQUEST_END);
-        ctx.log().addListener(RequestMetricSupport::onResponse,
+        ctx.log().addListener(log -> onResponse(log, meterIdPrefixFunction),
                               RequestLogAvailability.COMPLETE);
     }
 
     private static void onRequestStart(RequestLog log, MeterIdPrefixFunction meterIdPrefixFunction) {
         final RequestContext ctx = log.context();
         final MeterRegistry registry = ctx.meterRegistry();
         final MeterIdPrefix idPrefix = meterIdPrefixFunction.apply(registry, log);
-        final RequestMetrics requestMetrics =  MicrometerUtil.register(
-                registry, idPrefix, RequestMetrics.class, DefaultRequestMetrics::new);
+        final MeterIdPrefix idPrefixActive = new MeterIdPrefix(idPrefix.name("activeRequests"),
+                                                               idPrefix.tags());
+
+        final ActiveRequestMetrics activeRequestMetrics =  MicrometerUtil.register(
+                registry, idPrefixActive, ActiveRequestMetrics.class, DefaultActiveRequestMetrics::new);
 
-        ctx.attr(ATTR_REQUEST_METRICS).set(requestMetrics);
-        requestMetrics.active().incrementAndGet();
+        ctx.attr(ATTR_ACTIVE_REQUEST_METRICS).set(activeRequestMetrics);
```

Comment:
I'm curious if we really need to store `ActiveRequestMetrics` as a context attribute. We could just do this:

```java
// At setup():
.. Remove ctx.log().addListener(log -> onResponse(...)) ..

// At onRequestStart():
final ActiveRequestMetrics activeRequestMetrics = MicrometerUtil.register(...);
log.addListener(log -> onResponse(log, meterIdPrefixFunction, activeRequestMetrics), COMPLETE);

// At onResponse():
.. Do not access the context attribute at all, because we have direct refernece to `activeRequestMetrics` ..
```

As a result, the attribute value type of `ATTR_REQUEST_METRICS` could be just `Boolean` to determine if `setup()` was called before.